app/models/reservation.rb

```diff
@@ -189,7 +189,22 @@ def can_cancel?
   end
 
   def can_customer_edit?
-    !canceled? && !complete? && before_lock_window?
+    !canceled? && !complete? && (reserve_start_at_editable? || reserve_end_at_editable?)
+  end
+
+  def reserve_start_at_editable?
+    Time.zone.now < reserve_start_at && before_lock_window?
```

Comment:
This feels like it might be redundant. Wouldn't `before_lock_window?` always be true when `Time.zone.now < reserve_start_at`?